crates/bevy_app/src/app_builder.rs

```diff
@@ -177,37 +177,18 @@ impl AppBuilder {
         self
     }
 
-    pub fn on_state_enter<T: Clone + Component>(
-        &mut self,
-        stage: impl StageLabel,
-        state: T,
-        system: impl Into<SystemDescriptor>,
-    ) -> &mut Self {
-        self.stage(stage, |stage: &mut StateStage<T>| {
-            stage.on_state_enter(state, system)
-        })
+    pub fn add_state<T: Component + Clone + Eq>(&mut self, initial: T) -> &mut Self {
+        self.insert_resource(State::new(initial))
+            .add_system_set(State::<T>::make_driver())
```

Comment:
Is it possible to add these system sets (and hence states) to stages other than `Update`?

Reply:
`add_state_to_stage` is the solution for this.